### PR TITLE
Update sealing on new transactions

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

File: ethcore/src/miner/miner.rs [hunk 1]
```diff
@@ -564,7 +570,7 @@ impl MinerService for Miner {
 				for tx in &txs {
 					let _sender = tx.sender();
 				}
-				let _ = self.import_transactions(txs, |a| AccountDetails {
+				let _ = self.import_transactions(chain, txs, |a| AccountDetails {
```
Comment (tomusdrw) [inline ethcore/src/miner/miner.rs]: Sealing will be updated multiple times here:
1. Once for each retracted block + additional update sealing at the end of the function.

File: ethcore/src/client/client.rs [hunk 2]
```diff
@@ -771,7 +771,7 @@ impl<V> BlockChainClient for Client<V> where V: Verifier {
 			nonce: self.latest_nonce(a),
 			balance: self.latest_balance(a),
 		};
-		self.miner.import_transactions(transactions, fetch_account)
+		self.miner.import_transactions(self, transactions, fetch_account)
```
Comment (tomusdrw) [inline ethcore/src/client/client.rs]: This function (`BlockChainClient::import_transactions`) doesn't appear to be used anywhere - we only use `import_queued_transactions` now.

Comment (arkpar) [inline ethcore/src/client/client.rs]: It is still part of the client interface to import transactions synchronously. Could be useful for test.

File: ethcore/src/miner/miner.rs (hunk 1 above)
Comment (gavofyork) [inline ethcore/src/miner/miner.rs]: agree - could you file an optimisation issue?

**Comment by gavofyork** [comment]
can be merged after optimisation issue filed.

**Comment by gavofyork** [comment]
#1372.

File: ethcore/src/miner/miner.rs (hunk 1 above)
Comment (arkpar) [inline ethcore/src/miner/miner.rs]: #1373

**Pull request merged**